include/oneapi/dpl/pstl/onedpl_config.h

```diff
@@ -54,14 +54,34 @@
         but OpenMP headers are not found or the compiler does not support OpenMP"
 #endif
 
-#if (defined(SYCL_LANGUAGE_VERSION) || defined(CL_SYCL_LANGUAGE_VERSION)) &&                                           \
-    (__has_include(<sycl/sycl.hpp>) || __has_include(<CL/sycl.hpp>))
+// Preliminary check SYCL availability
+#if (__has_include(<sycl/sycl.hpp>) || __has_include(<CL/sycl.hpp>))
+#    define _ONEDPL_SYCL_HEADER_PRESENT 1
+#endif
+#if defined(CL_SYCL_LANGUAGE_VERSION) || defined(SYCL_LANGUAGE_VERSION)
+#    define _ONEDPL_SYCL_LANGUAGE_VERSION_PRESENT 1
+#endif
+#if _ONEDPL_SYCL_HEADER_PRESENT && _ONEDPL_SYCL_LANGUAGE_VERSION_PRESENT
 #    define _ONEDPL_SYCL_AVAILABLE 1
 #endif
-#if ONEDPL_USE_DPCPP_BACKEND && !_ONEDPL_SYCL_AVAILABLE
+// Intel(R) oneAPI DPC++/C++ Compiler provides SYCL_LANGUAGE_VERSION without sycl.hpp inclusion
+#if _ONEDPL_SYCL_HEADER_PRESENT && !_ONEDPL_SYCL_LANGUAGE_VERSION_PRESENT && !defined(__INTEL_LLVM_COMPILER)
```

Comment:
Will this cover the open-source intel/llvm implementation as well?

Reply:
It will. This implementation does not define `__INTEL_LLVM_COMPILER`

Reply:
So, the open source implementation enables SYCL in the same way as Intel(R) oneAPI DPC++/C++ Compiler (as far as I know), yet the logic in our header would work differently for them? Sounds strange to me.

Reply:
Let me think how to distinguish the open-sourced version from genetic clang++ to align the logic. Note that https://github.com/intel/llvm appears to have the same set of predefined macros as clang++, so we should rely on some indirect traits.

Reply:
`#if __has_cpp_attribute(intel::kernel_args_restrict)` or any other intel-specific attribute looks like an option to distinguish the compiler. Although, it does not look conventional.

Reply:
I've added a check with `#if __has_cpp_attribute(intel::kernel_args_restrict)`, but I am leaning towards removing it, as it looks non-conventinal and possibly fragile.

Reply:
This question is not relevant right now: all the compilers are treated equally.